pinot-core/src/test/java/org/apache/pinot/queries/JsonDatatypeTest.java

```diff
@@ -0,0 +1,293 @@
+  @Test
+  public void testExtractScalarWithArrayFilter() {
+    Operator operator = getOperatorForSqlQuery(
+        "select json_extract_scalar(jsonColumn, '$.name.first', 'STRING') FROM testTable WHERE json_extract_scalar(jsonColumn, '$.data[0]', 'STRING') IN ('i', 'k')");
+    IntermediateResultsBlock block = (IntermediateResultsBlock) operator.nextBlock();
+    Collection<Object[]> rows = block.getSelectionResult();
+    Assert.assertEquals(rows.size(), 2);
+
+    Iterator<Object[]> iterator = rows.iterator();
+    Assert.assertTrue(iterator.hasNext());
+    Assert.assertEquals(iterator.next()[0], "daisy");
+    Assert.assertEquals(iterator.next()[0], "goofy");
+  }
+
+  @Test
+  public void testJsonMatchWithoutIndex() {
+    try {
+      Operator operator = getOperatorForSqlQuery(
+          "select json_extract_scalar(stringColumn, '$.name.first', 'STRING') FROM testTable WHERE json_match(stringColumn, 'id=101')");
+      Assert.assertTrue(false);
+    } catch (IllegalStateException ise) {
+      Assert.assertTrue(true);
+    }
+  }
+
+  @Test
+  public void testJsonMatchAtLevel1WithIndex() {
+    Operator operator = getOperatorForSqlQuery(
+        "select json_extract_scalar(jsonColumn, '$.name.first', 'STRING') FROM testTable WHERE json_match(jsonColumn, 'id=101')");
+
+    IntermediateResultsBlock block = (IntermediateResultsBlock) operator.nextBlock();
+    Collection<Object[]> rows = block.getSelectionResult();
+    Assert.assertEquals(rows.size(), 1);
+
+    Iterator<Object[]> iterator = rows.iterator();
+    Assert.assertTrue(iterator.hasNext());
+    Assert.assertEquals(iterator.next()[0], "daffy");
+  }
+
+  @Test
+  public void testJsonMatchAtLevel2WithIndex() {
+    Operator operator = getOperatorForSqlQuery(
+        "select json_extract_scalar(jsonColumn, '$.name.first', 'STRING') FROM testTable WHERE json_match(jsonColumn, 'name.first = ''daffy''')");
+
+    IntermediateResultsBlock block = (IntermediateResultsBlock) operator.nextBlock();
+    Collection<Object[]> rows = block.getSelectionResult();
+    Assert.assertEquals(rows.size(), 1);
+
+    Iterator<Object[]> iterator = rows.iterator();
+    Assert.assertTrue(iterator.hasNext());
+    Assert.assertEquals(iterator.next()[0], "daffy");
+  }
+
+  @Test
+  public void testJsonMatchArrayWithIndex() {
+    Operator operator = getOperatorForSqlQuery(
+        "select json_extract_scalar(jsonColumn, '$.name.first', 'STRING') FROM testTable WHERE json_match(jsonColumn, '\"data[0]\" IN (''k'', ''j'')')");
+
+    IntermediateResultsBlock block = (IntermediateResultsBlock) operator.nextBlock();
+    Collection<Object[]> rows = block.getSelectionResult();
+    Assert.assertEquals(rows.size(), 2);
+
+    Iterator<Object[]> iterator = rows.iterator();
+    Assert.assertTrue(iterator.hasNext());
+    Assert.assertEquals(iterator.next()[0], "pluto");
+    Assert.assertTrue(iterator.hasNext());
+    Assert.assertEquals(iterator.next()[0], "goofy");
+  }
+
```

Comment:
Add a test for selecting a json column, and check the data schema of the response